src/app/problems/problem-view/java-view-snippet/java-view-snippet.component.ts

```diff
@@ -29,12 +30,14 @@ export class JavaViewSnippetComponent implements OnInit {
         this.inputFileNames.forEach(file => {
             codeSolution[file.name] = file.template;
         });
-        this.submissionService.postQuestionSubmission({question: this.uqj.question.id, solution: codeSolution})
+        this.submissionService.postQuestionSubmission({question: this.uqj.question.id, solution: codeSolution,
+            time_spent: parseInt(localStorage.getItem(this.uqj.id.toString()))})
             .subscribe(() => {
                 this.notificationsService
                     .show('The Question has been Submitted Successfully.', {
                         status: TuiNotification.Success
                     }).subscribe();
             });
+        localStorage.removeItem(this.uqj.id.toString());
```

Comment:
This may be better located in the `subscribe` may run into issues where it is removed if the submission does not work properly...

Reply:
Yeah, that's a good point, done.